app/perfunwind.cpp

```diff
@@ -682,6 +687,43 @@ void PerfUnwind::analyze(const PerfRecordSample &sample)
     sendBuffer(buffer);
 }
 
+void PerfUnwind::updateIpCache()
+{
+    const auto& stack = m_currentUnwind.ipStack;
+    auto& cache = m_ipCache[m_currentUnwind.sample->pid()];
+
+    // the entries in the cache don't need to be updated, since they already contain the longest known
+    // callchain for their ip -> only new ones need to be added
+
+    // limit length to prevent high memory usage since every new sample doubles the memory
+    // usage of the cache
+    const auto& frames = m_currentUnwind.frames.mid(0, std::min(1024, m_currentUnwind.ipStack.length()));
```

Comment:
this is super bogus, as it destroys COW - if at all this limit should be taken into account below in stitchCurrentCallChain's call to appendFramesTo to not grow the frames beyond a certain limit

Reply:
also rephrase the comment to indicate that this is a protection for the worst case, not for the common case. because normally, we would _not_ double the memory consumption for _every_ new sample?!